test/parallel/test-process-nice.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+
+if (common.isWindows || !common.isMainThread) {
+  assert.strictEqual(process.nice, undefined);
+  return;
+}
+
+assert.throws(
+  () => {
+    process.nice(NaN);
+  },
+  {
+    code: 'ERR_OUT_OF_RANGE',
+    name: 'RangeError [ERR_OUT_OF_RANGE]',
+    message: 'The value of "nice" is out of range. It must be ' +
+             'an integer. Received NaN'
+  }
+);
+
+assert.throws(
+  () => {
+    process.nice(Infinity);
+  },
+  {
+    code: 'ERR_OUT_OF_RANGE',
+    name: 'RangeError [ERR_OUT_OF_RANGE]',
+    message: 'The value of "nice" is out of range. It must be ' +
+             'an integer. Received Infinity'
+  }
+);
+
+[null, false, true, {}, [], () => {}, 'text'].forEach((val) => {
```

Comment:
How about also testing that passing '0' or `undefined` results in a finite integer being returned?

Reply:
Also, we might want to test that something like `process.nice(1)` works as expected.